lib/auth/register.go

```diff
@@ -611,6 +605,42 @@ func pinRegisterClient(params RegisterParams) (*Client, error) {
 	return authClient, nil
 }
 
+func caPathRegisterClient(params RegisterParams) (*Client, error) {
+	tlsConfig := utils.TLSConfig(params.CipherSuites)
+	tlsConfig.Time = params.Clock.Now
+
+	cert, err := readCA(params.CAPath)
+	if err != nil && !trace.IsNotFound(err) {
+		return nil, trace.Wrap(err)
+	}
+
+	// For backwards compatibility, we fall back to insecure mode if loading the caPath fails.
+	// TODO: At a future date we should remove this
```

Comment:
Ideally we have an owner and a rough idea of the future date.
```suggestion
	// DELETE IN 16 (strideynet)
```

Reply:
If we're happy to change the behaviour here, it'd definitely be nice to set a date and start warning more aggressively - but I do think we might need to do more investigation as for why it currently behaves the way it does. I really don't understand why the original implementation fails open?